fee/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use {
+    solana_sdk::fee::{FeeBudgetLimits, FeeDetails},
+    solana_svm_transaction::svm_message::SVMMessage,
+};
+
+/// Calculate fee for `SanitizedMessage`
+pub fn calculate_fee(
+    message: &impl SVMMessage,
+    lamports_per_signature: u64,
+    budget_limits: &FeeBudgetLimits,
+    remove_rounding_in_fee_calculation: bool,
+) -> u64 {
+    calculate_fee_details(
+        message,
+        lamports_per_signature,
+        budget_limits,
+        remove_rounding_in_fee_calculation,
+    )
+    .total_fee()
+}
+
+pub fn calculate_fee_details(
```

Comment:
@tao-stones following our discussion, I took a fresh start on fee-calculation function.

Simplifications:
1. Remove 0 lamports_per_signature edge-case
2. Remove write-lock fees (0 everywhere)
3. Remove CU fees (0 everywhere) - implicitly removes the effect of `include_loaded_account_data_size_in_fee`

If our tx/message type caches the total number of signatures this becomes a single multiplication and a few copies!

Reply:
Wonder how you feel about the following:

1. The only thing we use in `FeeBudgetLimits` is `prioritization_fee`. We could potentially just pass that instead.
2. If doing 1, we could also change the return type. 

Doing both of these removes this crates **explicit** dependency on `sdk`. An implicit dependence is still there due to `svm-transaction` using `sdk`.


```mermaid
graph LR;
sdk --> svm-transaction --> fee;
sdk -->|could remove?| fee
```

Reply:
Ended up not duplicated/deprecating `FeeBudgetDetails` because it does some additional serialization/deserialization and dcou stuff - would rather just not mess w/ it.